packages/instantsearch.js/src/templates/__tests__/carousel.test.tsx

```diff
@@ -0,0 +1,227 @@
+        navigationNext: 'NAVIGATION_NEXT',
+        navigationPrevious: 'NAVIGATION_PREVIOUS',
+      },
+    });
+
+    const { container } = render(
+      template({
+        items: [
+          { objectID: '1', __position: 1 },
+          { objectID: '2', __position: 2 },
+        ],
+        templates: {
+          item({ item }) {
+            return <p>{item}</p>;
+          },
+        },
+      })
+    );
+
+    expect(container.querySelector('.ais-Carousel')).toHaveClass('ROOT');
+    expect(container.querySelector('.ais-Carousel-list')).toHaveClass('LIST');
+    expect(container.querySelector('.ais-Carousel-item')).toHaveClass('ITEM');
+    expect(
+      container.querySelector('.ais-Carousel-navigation--previous')
+    ).toHaveClass('NAVIGATION', 'NAVIGATION_PREVIOUS');
+    expect(
+      container.querySelector('.ais-Carousel-navigation--next')
+    ).toHaveClass('NAVIGATION', 'NAVIGATION_NEXT');
+  });
+
+  test('adds custom templates', () => {
+    const template = carousel({
+      templates: {
+        previous() {
+          return <p>Previous</p>;
+        },
+        next() {
+          return <p>Next</p>;
+        },
+      },
+    });
+
+    const { container } = render(
+      template({
+        items: [
+          { objectID: '1', __position: 1 },
+          { objectID: '2', __position: 2 },
+        ],
+        templates: {
+          item({ item }) {
+            return <p>{item.objectID}</p>;
+          },
+        },
+      })
+    );
+
+    expect(container).toMatchInlineSnapshot(`
+      <div>
+        <div
+          class="ais-Carousel"
+        >
+          <button
+            aria-controls="ais-Carousel-2"
+            aria-label="Previous"
+            class="ais-Carousel-navigation ais-Carousel-navigation--previous"
+            hidden=""
+            title="Previous"
+          >
+            <p>
+              Previous
+            </p>
+          </button>
+          <ol
+            aria-label="Items"
+            aria-live="polite"
+            aria-roledescription="carousel"
+            class="ais-Carousel-list"
+            id="ais-Carousel-2"
+            tabindex="0"
+          >
+            <li
+              aria-label="1 of 2"
+              aria-roledescription="slide"
+              class="ais-Carousel-item"
+            >
+              <p>
+                1
+              </p>
+            </li>
+            <li
+              aria-label="2 of 2"
+              aria-roledescription="slide"
+              class="ais-Carousel-item"
+            >
+              <p>
+                2
+              </p>
+            </li>
+          </ol>
+          <button
+            aria-controls="ais-Carousel-2"
+            aria-label="Next"
+            class="ais-Carousel-navigation ais-Carousel-navigation--next"
+            title="Next"
+          >
+            <p>
+              Next
+            </p>
+          </button>
+        </div>
+      </div>
+    `);
+  });
```

Comment:
I don't think we need to add that or think about it yet, but when hit is supported, this needs insights tests and functionality too.

Reply:
Yes, we can accept `sendEvent` in there when it's time.